sycl/test-e2e/FilterSelector/select_device_hip.cpp

```diff
@@ -0,0 +1,62 @@
+// RUN: %{build} -o %t.out
+// RUN: env ONEAPI_DEVICE_SELECTOR=hip:gpu %t.out
```

Comment:
I need to replace "// RUN: env ONEAPI_DEVICE_SELECTOR=hip:gpu %t.out" with "// RUN: %{run} %t.out". Is that right ?

Reply:
> I don't see additional value in this test.

This is more important right now :)

Reply:
Do you mean there is no need to add this test for HIP ? Please explain your comment. Thanks.

Reply:
Exactly. Why aren't existing device selector tests enough?

Reply:
If there is a select device test for cuda, then there is a test for hip.

Reply:
No, your argument would be equivalent to saying that if you test "a + 1" and "a + 2" then there should also be a test for "a + 3". I think that would be a wrong claim.

Reply:
There is a test for opencl and another test for level zero. Is that right ?

Reply:
Do we need to argue for the addition of this test ?  "select_device_hip" contains the string "HIP" for testing the selection of HIP platform while "select_device_cuda" contains the string "CUDA" for the CUDA platform. Do you think they are equivalent ? If you think this test is not needed, please close the pull request.

Reply:
Personally, I'm OK with HIP test being added, saying that HIP, CUDA and L0 tests are identical bar the names and env variables, I wonder if it would be possible to merge them and have multiple run lines to check each case; not sure if extending the `REQUIRES` set would be worth it though?

Reply:
> +1 for merging. I don't think putting more conditions into `REQUIRES:` would work as all of them have to be satisfied at once.

Yes, that was my worry. This is very neat though:

> Instead, something like this _might_ be possible:
> RUN: %if cuda %{ ONEAPI_DEVICE_SELECTOR=cuda:gpu %{run-unfiltered-devices} ... %}
> ```